### ISSUE TITLE
Virtual Assistant Deployment - Multiple failures

### ISSUE THREAD

**Issue by ghost** [microsoft/botframework-solutions#2106]
**Document reference:** (Step 2) 
https://github.com/microsoft/botframework-solutions/blob/master/docs/tutorials/csharp/virtualassistant.md#deploy-your-assistant

See the screenshot:
![image](https://user-images.githubusercontent.com/7371750/62953560-5fa3d180-be0b-11e9-8b23-1f285e145c19.png)

I looked in the `deploy_cognitive_models_log.txt` log file and found entries saying that the script could not find files with `.luis` and `.qna` file extensions. These files are all there, only they are present with the `.lu` file extension.

For example:

Looking for |Actual file name
-------------|----------------------------------
`Deployment\Resources\LU\en\General.luis`|`General.lu`
`Deployment\Resources\QnA\en\Chitchat.qna`|`Chitchat.lu`
`Deployment\Resources\QnA\en\Faq.qna`|`Faq.lu`

Please fix.

**Comment by jwiley84** [comment]
Are you on a mac?

**Comment by ghost** [comment]
Windows 10.

**Comment by dfavretto** [comment]
Hi @sujayvsarma,

Could you share with us the command you are using for executing the script, along with the `deploy_cognitive_models_log.txt`? Also, by any chance, does your path to the `.lu` file has any space in it?

Regarding the message
```
Could not deploy LUIS model.
```
This error is commonly seen when there's already a LUIS model deployed with the same name, make sure there's no resources deployed. If you tried deploying before and it failed, you should have deleted all the resources before trying to deploy again.

**Comment by sujayvsarma** [comment]
This is the first time I've run the script, so I am quite sure there are no models etc preexisting.

I ran the deploy.ps1 script as instructed from the project directory. There are no spaces in the path - you can see my path in the error message in the attached screenshot in my original post above.

**Comment by ghost** [comment]
@dfavretto: 

1) The command is I ran `.\Deployment\Scripts\deploy.ps1` from PowerShell Core 6. I tried this in normal mode as well as in elevated mode, but the results are the same. 

2) Logs: [deploy_cognitive_models_log.txt](https://github.com/microsoft/botframework-solutions/files/3499886/deploy_cognitive_models_log.txt)

3) Existing LUIS model: None exist. I have verified. I also retried the deployment after clearing everything from Azure (including the App Registration in the AD). 

Now... 

The error message (visible in the above screenshot as well) says:

```
Error: Cannot find module 'C:\Users\Sujay'
```

That set me thinking. `C:\Users\Sujay` is not a valid path. My user profile's disk name is `Sujay Sarma`. And indeed the corresponding disk path is `C:\Users\Sujay Sarma`. This **has** a space in it. 

So basically the Node.js component is failing looking for something in my user profile folder? 

Can you guys fix this? Because wherever I login, Windows is going to create the exact same path again.

**Comment by dmvtech** [comment]
I'm looking at this currently. I should be able to reproduce and look into options.

**Comment by dfavretto** [comment]
With a more deeper observation on the screenshot you shared, and evaluating the deployment scripts, I see both times the error
```
Error: Cannot find module 'C:\Users\Sujay'
```
the command that was being executed is `LUDown`, so my initial guess would be that the cause of this issue is with that tool.

Adding to this, you are reporting and error which says the files that can't be found are the ones with extension `.luis` and `.qna`, and `LUDown` is the responsible of parsing the files with extension `.lu` to files with extension `.luis` and `.qna`, so that enforces my belief in `LUDown` being the cause of your problems.

Could you share with us the version of `LUDown` you have installed?

**Comment by ghost** [comment]
I have VS 2019 Community edition (on the preview branch, latest) + I installed the Bot Framework kit (the emulator installer) + the VSIX extension that installs the Virtual Assistant template (from VS marketplace).

I haven't installed any "ludown" manually.




regards.

**Comment by dmvtech** [comment]
@sujayvsarma Please see the link [here](https://microsoft.github.io/botframework-solutions/tutorials/csharp/create-assistant/2_download_and_install/), for the steps on installing ludown, etc tools that are needed for Virtual Assistant.

I am still looking into the issue regarding a space in the path.

**Comment by ghost** [comment]
@dmvtech - Where is the link?

**Comment by dmvtech** [comment]
Sorry about that, link added.

**Comment by ghost** [comment]
Can't you make all of those things into a single installer? Too many things to install !!!! And if something screws up, it's megapain. 

ps: for example, i have no idea how to fix npm installation issues.

**Comment by dmvtech** [comment]
@sujayvsarma I think they could be put into one package (all being dependents of that package). But there is current movement to move most/all these tools into one single tool, so would be really moot at that point. Please see [here](https://github.com/microsoft/botframework-cli) if you're curious.

**Comment by ghost** [comment]
> @sujayvsarma I think they could be put into one package (all being dependents of that package). But there is current movement to move most/all these tools into one single tool, so would be really moot at that point. Please see here if you're curious.

Sounds promising 👍

**Comment by ghost** [comment]
@dmvtech - I finally got the time to run the remaining commands. I ran it as a single line:

```
npm install -g botdispatch ludown luis-apis qnamaker luisgen@2.0.2 botskills
```

Installation was successful. However, I noticed during that installation that I somehow had "luisgen" version `2.2.0` already installed. The NPM command uninstalled that one and installed the (older) `2.0.2` version. If the scripts can use the newer `2.2.0` version, then the docs need to be updated.

**Comment by darrenj** [comment]
As per the documentation we recommend running the installation of botskills as a seperate line (hence it's provided that way) due to npm dependencies. It's really important to follow the pre-requisites as we bring together a number of technologies including az CLI.

Are you now able to complete deployment?

**Comment by ghost** [comment]
So do you want me to undo everything and try ? If so, I would need the relevant commands because I know nothing about NPM.

**Comment by darrenj** [comment]
You should be fine to be sure, run `npm uninstall -g botskills` and then run `npm install -g botskills`. You should then be good to go.

**Comment by darrenj** [comment]
Let us know if you still have issues after installing the pre-requsities as documented [here](https://microsoft.github.io/botframework-solutions/tutorials/csharp/create-assistant/2_download_and_install/)

**Comment by ghost** [comment]
Thank you for your help. I don't have time to continue this experiment at the moment. I shall do this later. I am closing this issue -- will open a new one if I run into further problems or need help.
